make.py

```diff
@@ -589,59 +528,13 @@ def run_clip_regression_test(cmd_queue, completed_queue, failed_queue, failure_l
 		if regression_testing_failed:
 			sys.exit(1)
 
-def print_help():
-	print('Usage: python make.py [actions] [cpu arch] [compiler] [config] [misc]')
-	print()
-	print('Actions:')
-	print('  If no action is specified, on Windows, OS X, and Linux the solution/make files are generated.')
-	print('  Multiple actions can be used simultaneously.')
-	print('  -build: Builds the solution.')
-	print('  -clean: Cleans the build directory.')
-	print('  -unit_test: Runs the unit tests.')
-	print('  -regression_test: Runs the regression tests.')
-	print()
-	print('CPU Architecture:')
-	print('  Only supported for Windows, OS X, and Linux. Defaults to the host system architecture.')
-	print('  Only a single architecture argument must be used.')
-	print('  -x86: Builds an x86 executable.')
-	print('  -x64: Builds an x64 executable.')
-	print()
-	print('Compiler:')
-	print('  Defaults to the host system\'s default compiler.')
-	print('  Only a single compiler argument must be used.')
-	print('  -vs2015: Uses Visual Studio 2015.')
-	print('  -vs2017: Uses Visual Studio 2017.')
-	print('  -osx: Uses X Code for OS X.')
-	print('  -gcc5: Uses GCC 5.')
-	print('  -gcc6: Uses GCC 6.')
-	print('  -gcc7: Uses GCC 7.')
-	print('  -gcc8: Uses GCC 8.')
-	print('  -clang4: Uses clang 4.')
-	print('  -clang5: Uses clang 5.')
-	print('  -clang6: Uses clang 6.')
-	print('  -android: Uses NVIDIA CodeWorks.')
-	print('  -ios: Uses X Code for iOS.')
-	print()
-	print('Config:')
-	print('  Defaults to Release.')
-	print('  Only a single config argument mus tbe used.')
-	print('  -debug: Uses the Debug configuration to build and test.')
-	print('  -release: Uses the Release configuration to build and test.')
-	print()
-	print('Miscelanous:')
-	print('  -avx: On Windows, OS X, and Linux AVX support will be enabled.')
-	print('  -help: Prints this help message.')
-
 if __name__ == "__main__":
-	options = parse_argv()
-	if options['print_help']:
-		print_help()
-		sys.exit(1)
+	args = parse_argv()
```

Comment:
How is help printed now?

Reply:
It's handled for you by argparse. The "help" action causes it to print out the following, then exit:

```
D:\Home Documents\Cody\Animation compression library\acl-test>py make.py -help
usage: make.py [-build] [-clean] [-unit_test] [-regression_test]
               [-compiler {vs2015,vs2017,android,clang4,clang5,clang6,gcc5,gcc6,gcc7,gcc8,osx,ios}]
               [-config {Debug,Release}] [-cpu {x86,x64}] [-avx] [-pop]
               [-nosimd] [-num_threads NUM_THREADS]
               [-tests_matching TESTS_MATCHING] [-help]

Actions:
  If no action is specified, on Windows, OS X, and Linux the solution/make
  files are generated. Multiple actions can be used simultaneously.

  -build
  -clean
  -unit_test
  -regression_test

Target:
  -compiler {vs2015,vs2017,android,clang4,clang5,clang6,gcc5,gcc6,gcc7,gcc8,osx,ios}
                        Defaults to the host system's default compiler
  -config {Debug,Release}
  -cpu {x86,x64}        Only supported for Windows, OS X, and Linux; defaults
                        to the host system's architecture

Miscellaneous:
  -avx                  Compile using AVX instructions on Windows, OS X, and
                        Linux
  -pop                  Compile using the POPCNT instruction
  -nosimd               Compile without SIMD instructions
  -num_threads NUM_THREADS
                        No. to use while compiling and regressing
  -tests_matching TESTS_MATCHING
                        Only run tests whose names match this regex
  -help                 Display this usage information
```